src/common/functions/searchProf.ts

```diff
@@ -35,4 +41,18 @@ export function searchProf(
       @@ to_tsquery(${processedQuery + ":*"})
     LIMIT ${limit};
   `;
+
+  return Promise.all(
+    queryResult.map(async (r) => {
+      const [courseCount, reviewCount] = await Promise.all([
+        await api.courses.countByProfSlug({ slug: r.profSlug }),
+        await api.reviews.countByProfessorSlug({ slug: r.profSlug }),
+      ]);
+      return {
+        ...r,
+        courseCount,
+        reviewCount,
+      };
+    }),

```

Comment:
thought: this repetitive db query is done to re-use premade functions  instead of building (and maintaining) a larger SQL query for search - as a bandaid solution

if this query becomes too slow or impact to db is too large, we should consider merging the query with that in ln 28